docs/tools/firmware.sh

```diff
@@ -18,36 +18,178 @@ while getopts "vhx" option; do
-		echo "Unmounting the EFI partition"
-		sudo diskutil unmount "/Volumes/${EFILABEL}/"
-		echo
-		echo -e "Run the following commands or run this script itself in Linux now to set up Wi-Fi :-\n\nsudo mkdir -p /tmp/apple-wifi-efi\nsudo mount /dev/nvme0n1p1 /tmp/apple-wifi-efi\nbash /tmp/apple-wifi-efi/firmware.sh\nsudo umount /tmp/apple-wifi-efi\n"
+					nvramfile=$(ioreg -l | grep RequestedFiles | cut -d "/" -f 5 | rev | cut -c 4- | rev)
+					txcapblob=$(ioreg -l | grep RequestedFiles | cut -d "/" -f 3 | cut -d "\"" -f 1)
+					cp ${verbose} /usr/share/firmware/wifi/C-4364__s-B2/${nvramfile} "/Volumes/${EFILABEL}/brcmfmac4364b2-pcie.txt"
+					cp ${verbose} /usr/share/firmware/wifi/C-4364__s-B2/${txcapblob} "/Volumes/${EFILABEL}/brcmfmac4364b2-pcie.txcap_blob"
+				fi
+				echo "Copying this script to EFI"
+				cp "$0" "/Volumes/${EFILABEL}/firmware.sh" 2>/dev/null || curl -s https://wiki.t2linux.org/tools/firmware.sh > "/Volumes/${EFILABEL}/firmware.sh" || (echo -e "\nFailed to copy script.\nPlease copy the script manually to the EFI partition using Finder\nMake sure the name of the script is firmware.sh in the EFI partition\n" && echo && read -p "Press enter after you have copied" && echo)
+				echo "Unmounting the EFI partition"
+				sudo diskutil unmount "/Volumes/${EFILABEL}/"
+				echo
+				echo -e "Run the following commands or run this script itself in Linux now to set up Wi-Fi :-\n\nsudo mkdir -p /tmp/apple-wifi-efi\nsudo mount /dev/nvme0n1p1 /tmp/apple-wifi-efi\nbash /tmp/apple-wifi-efi/firmware.sh\nsudo umount /tmp/apple-wifi-efi\n"
+				;;
+			(2)
+
+				echo -e "\nPrerequisites needed to continue:"
+				echo -e "\n1. Python 3"
+				echo -e "\nPython 3 is a part of Xcode command line tools. You can also use Homebrew, MacPorts etc."
+				echo "Continue when you have the prerequisites installed."
+				echo -e "\nContinue? (Y/n)"
+				read input
+				if [[ ($input = n) || ($input = N) ]]
+				then
+					echo -e "\nRun the script again when you have installed the prerequisites."
+					exit 1
+				fi
+				echo -e "\nCreating tar archive of the firmware"
+				python3 $0 /usr/share/firmware $HOME/Downloads/firmware.tar ${verbose}
+				echo -e "\nFirmware tar archive saved to Downloads!"
+				echo -e "\nExtract the firmware contents to /lib/firmware/brcm in Linux and run the following in the Linux terminal:"
+				echo -e "\nsudo modprobe -r brcmfmac_wcc"
+				echo "sudo modprobe -r brcmfmac"
+				echo "sudo modprobe brcmfmac"
+				echo "sudo modprobe -r hci_bcm4377"
+				echo "sudo modprobe hci_bcm4377"
+				;;
+			(3)
+				echo -e "\nWhat package manager does your Linux distribution use?"
+				echo -e "\n1. apt"
+				echo "2. rpm"
+				read package
+				case ${package} in
+					(1)
+						additional_requirement=dpkg
+						prerequisite_packages
+						read input
+						if [[ ($input = n) || ($input = N) ]]
+						then
+							echo -e "\nRun the script again when you have installed the prerequisites."
+							exit 1
+						fi
```

Comment:
Make `prerequisite_packages` return a non-zero exit status if dependencies are not installed, then remove this code and replace it with `prerequisite_packages || exit 1`